app/models/user.rb

```diff
@@ -116,6 +116,8 @@ def pending_profile
     ).order(created_at: :desc).first
 
     return unless pending.present?
+    return if pending.password_reset?
+    return if pending.encryption_error? || pending.verification_cancelled?
```

Comment:
Should these be clauses in the SQL query? It wouldn't be an issue unless the query above could match multiple Profiles, but it may be worth being more defensive?

I happened to be messing with this due to the bug, and I don't think it's straightforward to do it via ActiveRecord, so I ended up with something like:

```ruby
pending = profiles.where(
  <<-SQL
  (deactivation_reason IS NULL OR deactivation_reason != 2) AND
  (deactivation_reason = 5 OR gpo_verification_pending_at IS NOT NULL OR fraud_review_pending_at IS NOT NULL OR fraud_rejection_at IS NOT NULL)
  SQL
).order(created_at: :desc).first
```

Reply:
> Should these be clauses in the SQL query? It wouldn't be an issue unless the query above could match multiple Profiles, but it may be worth being more defensive?

Yeah, I messed around with this a bit trying to get the ActiveRecord query to work. But, I do think we want to return nil if the candidate pending profile was cancelled or reset instead of the next potentially pending profile in line.

Reply:
Because, ultimately, a user should only ever have 1 pending profile. If they cancel their pending profile then the query should not be able to find that pending profile. The intent is they proof again or deal with their password reset profile.

Reply:
`Profile.deactivation_reasons` is a hash with indifferent access of the reasons, so we could put in the enum values like this:

```ruby
profiles.where(format(<<-SQL), Profile.deactivation_reasons)
  (deactivation_reason IS NULL OR deactivation_reason != %{encryption_error}) AND
  (deactivation_reason = %{in_person_verification_pending} OR gpo_verification_pending_at IS NOT NULL OR fraud_review_pending_at IS NOT NULL OR fraud_rejection_at IS NOT NULL)
SQL
```